### PR TITLE
Thread time statistics

### PR DESCRIPTION
**Pull request by andyp1per**
If CH_DBG_STATISTICS is defined in hwdef.dat then additional thread statistics are added to threads.txt:
```
ThreadsV2
ISR           PRI=255 sp=0x20010000 STACK=1320/1536
ArduCopter    PRI=182 sp=0x20010600 STACK=6380/8192 MIN=  72 AVG=1799 MAX=23188
idle          PRI=  1 sp=0x200227F8 STACK= 324/ 480 MIN= 156 AVG=2105 MAX=40657
apm_uart      PRI= 60 sp=0x200240B8 STACK=2096/2512 MIN=  73 AVG= 885 MAX=2755
monitor       PRI=183 sp=0x2001CDC0 STACK= 848/1232 MIN= 176 AVG= 265 MAX=1089
timer         PRI=181 sp=0x2001E250 STACK=2148/2512 MIN=  71 AVG=1067 MAX=2822
rcin          PRI=177 sp=0x2001D2F0 STACK= 696/1232 MIN=  74 AVG= 459 MAX=2939
io            PRI= 58 sp=0x2001C390 STACK=1588/2512 MIN=  70 AVG= 596 MAX=21202
storage       PRI= 59 sp=0x2001D820 STACK=2008/2512 MIN=  80 AVG= 323 MAX=3537
I2C0          PRI=176 sp=0x20028BD8 STACK= 960/1488 MIN=  82 AVG= 621 MAX=1685
SPI4          PRI=181 sp=0x20038B68 STACK= 612/1488 MIN=  72 AVG=1010 MAX=4672
FTP           PRI= 58 sp=0x2003C378 STACK=2128/3536 MIN=  81 AVG= 337 MAX=5138
```

### CODE REVIEW

**Review by khancyr — APPROVED** [review]
Looks good

**Comment by tridge** [comment]
for some reason I didn't think that worked in tickless mode (we run ChibiOS tickless)

**Comment by andyp1per** [comment]
@tridge it works fine in tickless - it's thread profiling that doesn't work in tickless. Look at the last paragraph on https://www.chibios.org/dokuwiki/doku.php?id=chibios:documentation:books:rt:kernel_debug

File: libraries/AP_Filesystem/AP_Filesystem_Sys.cpp [hunk 1]
```diff
@@ -31,7 +31,11 @@ struct SysFileList {
 };
 
 static const SysFileList sysfs_file_list[] = {
+#ifdef CH_DBG_STATISTICS
```
Comment (tridge) [inline libraries/AP_Filesystem/AP_Filesystem_Sys.cpp:34]: #if HAL_ENABLE_THREAD_STATISTICS
#endif

and default that define based on CH_DBG_STATISTICS TRUE

**Review by tridge — CHANGES_REQUESTED** [review]

File: libraries/AP_Filesystem/AP_Filesystem_Sys.cpp (hunk 1 above)
Comment (tridge) [reply libraries/AP_Filesystem/AP_Filesystem_Sys.cpp:34]: it would be nice to have a way to reset it

Comment (andyp1per) [reply libraries/AP_Filesystem/AP_Filesystem_Sys.cpp]: Fixed

**Review by tridge — APPROVED** [review]

**Pull request merged**